client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx

```diff
@@ -103,15 +144,68 @@ const SiteSetupList = ( {
 	tasks,
 	taskUrls,
 	userEmail,
+	firstIncompleteTask,
 } ) => {
-	const [ currentTaskId, setCurrentTaskId ] = useState( null );
-	const [ currentTask, setCurrentTask ] = useState( null );
-	const [ userSelectedTask, setUserSelectedTask ] = useState( false );
-	const [ useDrillLayout, setUseDrillLayout ] = useState( false );
-	const [ currentDrillLayoutView, setCurrentDrillLayoutView ] = useState( 'nav' );
-	const [ isLoading, setIsLoading ] = useState( false );
+	// Data layer dispatch
 	const dispatch = useDispatch();
 
+	// Local state reducer
+	const [ state, localDispatch ] = useReducer( siteSetupListReducer, {
+		currentTaskId: null, // id of the current task
+		currentTask: null, // full object for current task
+		userSelectedTask: false, // was the currently selected task selected by the user
+		useDrillLayout: false, // is using small-screen layout view
+		currentDrillLayoutView: 'nav', // viewing a card on the list of checklist steps?
+	} );
+
+	// Set currentTaskId
+	const currentTaskId = state.currentTaskId;
+	const setCurrentTaskId = ( id ) =>
+		localDispatch( {
+			type: 'SET_CURRENT_TASK_ID',
+			currentTaskId: id,
+		} );
+
+	// Set currentTask
+	const currentTask = state.currentTask;
+	const setCurrentTask = ( task ) =>
+		localDispatch( {
+			type: 'SET_CURRENT_TASK',
+			currentTask: task,
+		} );
+
+	// Set userSelectedTask
+	const userSelectedTask = state.currentTask;
+	const setUserSelectedTask = ( isUserSelectedTask ) =>
+		localDispatch( {
+			type: 'SET_USER_SELECTED_TASK',
+			userSelectedTask: isUserSelectedTask,
+		} );
+
+	// Set useDrillLayout
+	const useDrillLayout = state.useDrillLayout;
+	const setUseDrillLayout = ( shouldUseDrillLayout ) =>
+		localDispatch( {
+			type: 'SET_USE_DRILL_LAYOUT',
+			useDrillLayout: shouldUseDrillLayout,
+		} );
+
+	// Set currentDrillLayoutView
+	const currentDrillLayoutView = state.currentDrillLayoutView;
+	const setCurrentDrillLayoutView = ( drillLayoutView ) =>
+		localDispatch( {
+			type: 'SET_CURRENT_DRILL_LAYOUT_VIEW',
+			currentDrillLayoutView: drillLayoutView,
+		} );
+
+	// Set isLoading
+	const isLoading = state.isLoading;
+	const setIsLoading = ( currentlyLoading ) =>
+		localDispatch( {
+			type: 'SET_IS_LOADING',
+			isLoading: currentlyLoading,
+		} );
```

Comment:
Seems like a lot of boilerplate, was the previous component state getting hard to follow?

Reply:
It was added so that I could pass the local reducer's `dispatch` into callbacks. However that requirement no longer exists so I've reverted all the changes to state back to how they were originally.